src/proof_rules/mod.rs

```diff
@@ -97,6 +126,18 @@ pub trait Encoding: fmt::Debug {
     /// Check if the given calculus annotation is compatible with the encoding annotation
     fn is_calculus_allowed(&self, calculus: Calculus, direction: Direction) -> bool;
 
+    /// Select an approximation kind based on the semantics type used.
+    fn get_approximation(
+        &self,
+        fixpoint_semantics: FixpointSemanticsKind,
+        inner_approximation_kind: ApproximationKind,
+    ) -> ApproximationKind;
+
+    /// Get the sound fixpoint semantics kind for this encoding annotation based on the direction
```

Comment:
Why "sound"? I think the name should just be `default_fixpoint_semantics()` or something similar.